lib/cartopy/tests/io/test_ogc_clients.py

```diff
@@ -169,6 +174,33 @@ def test_fetch_img(self):
         self.assertEqual((-180.0, 107.99999999999994,
                           -197.99999999999994, 90.0), located_image.extent)
 
+    def test_fetch_img_reprojected(self):
+        source = ogc.WMTSRasterSource(self.URI, self.layer_name)
+        extent = [-20, -1, 48, 50]
+        # NB single result in this case.
+        located_image, = source.fetch_raster(ccrs.NorthPolarStereo(), extent,
+                                             (30, 30))
+
+        # Check image array is as expected (more or less).
+        img = np.array(located_image.image)
+        self.assertEqual(img.shape, (42, 42, 4))
+        # When reprojected, extent is exactly what you asked for.
+        self.assertEqual(located_image.extent, extent)
+
+    def test_fetch_img_reprojected_twoparts(self):
+        source = ogc.WMTSRasterSource(self.URI, self.layer_name)
+        extent = [-10, 12, 48, 50]
+        # NB expect 2 results in this case.
```

Comment:
If you expect two results, then it's better to assert that. If the length is not two, then tuple unpacking will result in an error instead of a failure.